### PR TITLE
Documentation: Overview

### PR DESCRIPTION
**Pull request by manonthemat**
I've added an overview section in the readme which I think makes it easier to see what error codes boom currently covers. Another benefit is that navigating to the appropriate section should be faster as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -491,6 +525,26 @@ Generates the following response payload:
 
 All 500 errors hide your message from the end user. Your message is recorded in the server log.
 
+### `Boom.badImplementation([message], [data])`
```
Comment (arb) [inline README.md]: Why the move here?

Comment (manonthemat) [inline README.md]: I sorted the methods by http status code in ascending order, thus resulting in the move here.

**Comment by arb** [comment]
Since you opened this PR, preconditionRequired, was added. Can you make sure it's in the right place in the documentation after you pull from upstream?

**Comment by manonthemat** [comment]
Now it is.

**Comment by arb** [comment]
Thanks!

**Pull request merged**